### DISCUSSION TITLE
Artifact update problem for private GitHub repository dependency

### DISCUSSION THREAD

**Discussion by schw4rzlicht** [renovatebot/renovate#14571] [Request Help]
Hey there,

some key facts:
* we use the GitHub app
* we use npm
* we have a private GitHub repository listed as dependency in our `package.json`. It looks like this: `"package-name": "git+https://github.com/vendor/package.git#<commitSHA>"`.
* renovate app is authorized to read `vendor/package` repository
* `vendor` is a different organization than the consuming repository belongs to (if that matters)

In my opinion, everything should be configured correctly but still we get this log posted to our PRs:

```
npm WARN using --force I sure hope you know what you are doing.
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN using --force I sure hope you know what you are doing.
chmod: cannot access '/home/ubuntu/.npm': No such file or directory

[snipped some package deprecation notices]

npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t https://github.com/vendor/package.git
npm ERR! 
npm ERR! remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
npm ERR! remote: Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
npm ERR! fatal: Authentication failed for 'https://github.com/vendor/package.git/'
npm ERR! 
npm ERR! exited with error code: 128

npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/renovate-cache/others/npm/_logs/2022-03-07T00_46_42_631Z-debug.log
```

I am not really sure how to work around that. What we already do is using private npm packages from the GitHub registry. For that we configured an encrypted `npmToken` and `npmrc` entry in `renovate.json`. This used to work and still works.

My idea was to use the same token for git auth but I cannot really figure out how. Especially since I thought it should already work, given that renovate has access to `vendor/package`.

Thanks for help and regards
Julian

**Comment by rarkins** [comment]
I think this would require integration with https://github.com/renovatebot/renovate/blob/main/lib/util/git/auth.ts

**Reply by schw4rzlicht** [reply]
Could you please elaborate on that a bit more?

**Reply by rarkins** [reply]
Renovate runs npm in a "sidecar" image. The image itself doesn't know any auth details unless we tell it. In this case we'd need to pass auth details to git, because npm calls git. This is already done for some package managers using the above auth file/functions.
